src/Serializer/Filter/PropertyFilter.php

```diff
@@ -129,18 +134,42 @@ private function getProperties(array $properties, array $whitelist = null): arra
 
         foreach ($properties as $key => $value) {
             if (is_numeric($key)) {
-                if (\in_array($value, $whitelist, true)) {
-                    $result[] = $value;
+                if (\in_array($propertyName = $this->denormalizePropertyName($value), $whitelist, true)) {
+                    $result[] = $propertyName;
                 }
 
                 continue;
             }
 
             if (\is_array($value) && isset($whitelist[$key]) && $recursiveResult = $this->getProperties($value, $whitelist[$key])) {
-                $result[$key] = $recursiveResult;
+                $result[$this->denormalizePropertyName($key)] = $recursiveResult;
+            }
+        }
+
+        return $result;
+    }
+
+    private function denormalizeProperties(array $properties): array
+    {
+        if (null === $this->nameConverter || !$properties) {
+            return $properties;
+        }
+
+        $result = [];
+        foreach ($properties as $key => $value) {
+            if (is_numeric($key)) {
+                $result[] = $this->denormalizePropertyName($value);
```

Comment:
note for self, need to check here that `$value` is a string if not done before so it would not blow with strict typing

Reply:
wouldn't it be easier to simplify the method `denormalizePropertyName` so that it accept anything? It'd be compliant with the name converter denormalize method and avoid these checks.

Reply:
i'll do that ! the code will be simpler :+1:

Reply:
updated, thanks for pointing that out